Add `access`

https://pubs.opengroup.org/onlinepubs/9799919799/functions/access.html

Minimal example:

```C
#include <unistd.h>

int main() {
  access("", 0);
}
```

```
root@7434de944956 /mnt [1]# x86_64-hermit-gcc -o main main.c -L/mnt
/opt/hermit/lib/gcc/x86_64-hermit/14.2.0/../../../../x86_64-hermit/bin/ld: /tmp/ccUWIXi4.o: in function `main':
main.c:(.text+0xf): undefined reference to `access'
collect2: error: ld returned 1 exit status
```

--- a/newlib/libc/sys/hermit/unistd.c
+++ b/newlib/libc/sys/hermit/unistd.c
@@ -3,6 +3,7 @@
 #include <pthread.h>
 #include <stdint.h>
 #include <string.h>
+#include <sys/stat.h>
 #include <time.h>
 #include <unistd.h>
 
@@ -10,6 +11,27 @@ void sys_exit(int status);
 
 void _exit(int status) {
 	sys_exit(status);
+}
+
+int access(const char *path, int amode) {
+	struct stat s;
+	if (stat(path, &s)) {
+		return -1;
+	}
+
+	if (s.st_mode & S_IFDIR) {
+		return 0;
+	}
+	
+	if (amode & W_OK) {
+		if (s.st_mode & S_IWRITE) {
+			return 0;
+		}
+
+		return -1;
+	}
+
+	return 0;
 }
 
 int chown(const char *path, uid_t owner, gid_t group) {

